test/unit/slot/distribution.js

```diff
@@ -54,6 +54,69 @@ describe('slot/distribution', () => {
     expect(frag.childNodes.length).to.equal(0);
   });
 
+  describe('changing slot name with three children', () => {
```

Comment:
three specifically, nothing. Just had an idea that pulling things out and putting them back in could break things, so added a test for it which showed the bug on line 339
https://github.com/skatejs/named-slots/pull/111#discussion-diff-73829795L339